app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb

```diff
@@ -206,19 +257,36 @@ def resource_uri
     ].join('/')
   end
 
-  def api_counters_info
-    COUNTERS_INFO[:api]
+  def resource_url(tail_path = nil, query = {})
+    @resource_url ||= URI.join(@connection.environment.resource_url, resource_uri).freeze
+
+    url       = @resource_url.dup
+    url.path  = [url.path, tail_path].join('/') if tail_path
+    url.query = URI.encode_www_form(query) unless query.empty?
+
+    url
   end
 
-  def api_metric_names
-    METRIC_NAMES[:api]
+  def metrics_url
+    @metrics_url ||= resource_url(
+      # https://docs.microsoft.com/en-us/rest/api/monitor/metrics/list#uri-parameters
+      'providers/microsoft.insights/metrics',
+      'timespan'    => "#{@start_time}/#{@end_time}",
+      'interval'    => INTERVAL_1_MINUTE,
+      'metricnames' => METRIC_NAMES[:api].join(','),
+      'aggregation' => 'Average',
+      'api-version' => '2018-01-01'
+    ).freeze
   end
 
-  def raw_counters_info
-    COUNTERS_INFO[:raw]
+  def definitions_url
+    @definitions_url ||= resource_url(
+      'providers/microsoft.insights/metricDefinitions',
+      'api-version' => '2015-07-01'
```

Comment:
Do we want to hard code this? I can't remember, but I thought we set in this in config/settings.yml.

Reply:
Honestly, I'm surprised by this question. Why do we hardcode gem versions everywhere? It is the same - every api version has it's own differences, and this code relies on the exact specific version.

Reply:
I don't really expect people to tinker with the api-version strings in the settings, it's just a matter of having it in one place instead of (potentially) many places throughout the code for convenience. Also, MS will completely drop support for api versions after a while, but I guess we could just say this particular release is no longer supported then.

Reply:
@djberg96 so I am playing with this in Topo and I've decided to just use version `Latest`, since this is still tied to the gem version, correct? 

So the Latest will keep pointing to the same version, until we update the gem. And i'd expect they don't add new versions with patch release?

Reply:
@Ladas that is essentially the default behavior of the azure-armrest gem, but we lock in api-versions in ManageIQ because sometimes MS breaks things between versions.

Reply:
@djberg96 right, my point was: isn't it enough to specify gem version, which contains only specific API versions (as classes)? (we'd have to move to latest in the gem though)

Also I am using the Azure's official gem in topo, so not sure how arm-rest differs there.

Reply:
No, the azure-armrest gem itself is not tied to specific api-version strings (in most cases). It will generally try to use the latest, or latest preview:

https://github.com/ManageIQ/azure-armrest/blob/master/lib/azure/armrest/configuration.rb#L266-L286

Reply:
See also: https://github.com/ManageIQ/azure-armrest/wiki/Locking-Down-the-%22api-version%22-Strings

Reply:
Version, mentioned here - is the most recent, which supports table data info. Previously it was: https://github.com/ManageIQ/azure-armrest/blob/42d8728294643757d7f86f208a9a78100eae349f/lib/azure/armrest/insights/metrics_service.rb#L129
Anyway, I see all this code as an intermediate step on the road to Azure-own gemset instead of `azure-armrest`.

Reply:
Ah ok, the official gem has actual versioned classes in the gem.